File: 1371-find-the-longest-substring-containing-vowels-in-even-counts/1371-find-the-longest-substring-containing-vowels-in-even-counts.java
class Solution {
    public int findTheLongestSubstring(String s) {
        int bitmask=0;
        int max=0;
        HashMap<Integer,Integer> hm=new HashMap<>();
        hm.put(0,-1);
        for(int i=0;i<s.length();i++){
            char c=s.charAt(i);
            if(c=='a' || c=='e' || c=='o' || c=='i' || c=='u')
              bitmask=bitmask^(1<<(c-'a'));
            if(!hm.containsKey(bitmask))    
                hm.put(bitmask,i);
            max=Math.max(max,i-hm.getOrDefault(bitmask,0));

        }
        return max;
    }
}

File: 1371-find-the-longest-substring-containing-vowels-in-even-counts/1371-find-the-longest-substring-containing-vowels-in-even-counts.cs
public class Solution {
    public int FindTheLongestSubstring(string s) {
        Dictionary<int,int> dict=new Dictionary<int,int>();
        int bitmask=0;
        int max=0;
        dict.Add(0,-1);
        
        for(int i=0;i<s.Length;i++){
            char c=s[i];
            if(c=='a' || c=='e' || c=='i' || c=='o' || c=='u')
                bitmask=bitmask^(1<<(c-'a'));
            if(!dict.ContainsKey(bitmask))
                dict.Add(bitmask,i);
            int def;
            max=Math.Max(max,i-(dict.TryGetValue(bitmask,out def) ? def : 0));
        }
        return max;
    }
}